README.md

```diff
@@ -2,49 +2,62 @@
-
-**The autoscaler** - responsible to periodically check whether resources should be scale to zero, it does that by 
-querying the custom metrics API. When it decide a resource should be scaled to zero it tells the resource scaler to 
-scale the resource to zero.
-The resource scaler will route the traffic to the DLX, which in terms of K8s is done by changing the svc selector, 
-after that, it will scale the resource to zero.
-
-**The DLX** - responsible for receiving and buffering requests of scaled to zero resources, when it gets a message it 
-creates a buffer for the messages, and tells the resource scaler to scale the service back from zero.
-The resource scaler will scale the resource back up and then route the traffic back to the service (by changing the svc 
-selector).
+but you can use which ever you want! You can find some recommended implementations 
+[here](https://github.com/kubernetes/metrics/blob/master/IMPLEMENTATIONS.md#custom-metrics-api)
 
 ## Getting Started
 This infrastructure designed to be generic and extendable, it can scale any resource.
-All you have to do is implement the resource scaler, the interface between it and this infrastructure's components is 
-defined in [scaler-types](https://github.com/v3io/scaler-types)
+All you have to do is implement the specific resource-scaler for your resource. The interface between your 
+resource-scaler and the scale-to-zero infrastructure's components is defined in 
+[scaler-types](https://github.com/v3io/scaler-types)
 
-**Note:** Incompatibility between the scaler repo vendor dir and the resource scale vendor dir may break things, 
-therefore it's suggested to put the resource scaler in its own repo
+**Note:** Incompatibility between this scaler vendor dir and your resource-scale vendor dir may break things, 
+therefore it's suggested to put your resource-scaler in its own repo
 
 Examples:
-* [Nuclio functions resource scaler](https://github.com/nuclio/nuclio/blob/master/pkg/platform/kube/resourcescaler/resourcescaler.go)
-* [Iguazio's app service resource scaler](https://github.com/v3io/app-resource-scaler/blob/development/resourcescaler.go)  
+* [Nuclio functions resource-scaler](https://github.com/nuclio/nuclio/blob/master/pkg/platform/kube/resourcescaler/resourcescaler.go)
+* [Iguazio's app service resource-scaler](https://github.com/v3io/app-resource-scaler/blob/development/resourcescaler.go)  
 
 ## Installing
-Go plugin is the magic that glues the resource scaler and this infrastructure components together
-First you'll need to build a Dockerfile that builds your resource scaler as a Go plugin, and transplant it in this 
-repo released images
-
-Here are some great examples:
+[Go plugins](https://appliedgo.net/plugins/) is the magic that glues the resource-scaler and this infrastructure 
+components together.<br>
+First you'll need to build the resource scaler as a Go plugin, for example: <br>
```

Comment:
`resource scaler` -> `resource-scaler` everywhere then :) (since you went with this suggestion in some places make it consistent)